### PR TITLE
Hide source from dropdown if already selected

### PR DESCRIPTION
**Pull request by Shabanisimo**
https://pathcheck.atlassian.net/browse/SAF-195
photo_2020-05-08_12-33-57
photo_2020-05-08_12-34-00

### CODE REVIEW

File: app/views/ChooseProvider.js [hunk 1]
```diff
@@ -367,16 +368,19 @@ class ChooseProviderScreen extends Component {
...
-                      key={key}
-                      onSelect={() => {
-                        this.addAuthorityToState(name);
-                      }}
-                      disabled={this.state.authoritiesList.length === 1}>
-                      <Typography style={styles.menuOptionText} use={'body2'}>
-                        {name}
-                      </Typography>
-                    </MenuOption>
+                    _.findIndex(this.state.selectedAuthorities, ['key', name]) <
```
Comment (Patrick-Erichsen) [inline app/views/ChooseProvider.js]: So - this whole section was pretty messy before you arrived - but do you think you could pull it out into a new component, something like `<AuthoritiesMenu />`?  Basically lines 348 - 397. Could also have a few other distinct components in there, e.g. `<GPSFilter />`, `<AuthoritiesMenuOption />`

I know that it's a little out-of-scope with the original ticket, but it feels like a good opportunity to get this cleaned up.

Comment (Patrick-Erichsen) [reply app/views/ChooseProvider.js]: Also - could you pull the `_.findIndex()` method into a separate method, something like `isAuthoritySelected()`? It's a little hard to understand what that code is doing at a quick glance.

File: app/views/ChooseProvider.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import _ from 'lodash';
```
Comment (tstirrat) [inline app/views/ChooseProvider.js]: Use the function import instead of the whole library:

`import findIndex from 'lodash/findIndex';`

File: app/views/ChooseProvider.js (hunk 1 above)
Comment (Shabanisimo) [reply app/views/ChooseProvider.js]: > Also - could you pull the `_.findIndex()` method into a separate method, something like `isAuthoritySelected()`? It's a little hard to understand what that code is doing at a quick glance.
Okey, I'll update this component.

**Comment by Patrick-Erichsen** [comment]
Hye @Shabanisimo , thanks for doing some of that cleanup work! Please fix the merge conflict and this is good to go.

File: app/components/TrustedSourceList.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import findIndex from 'lodash/findIndex';
+import React from 'react';
+import { StyleSheet, Switch, TouchableOpacity } from 'react-native';
+import { MenuOption } from 'react-native-popup-menu';
+
+import { Typography } from '../components/Typography';
+import Colors from '../constants/colors';
+import languages from '../locales/languages';
+
+export default function TrustedSourceList({
```
Comment (tstirrat) [inline app/components/TrustedSourceList.js]: use a named export, not default.

File: app/views/LocationTracking.js [hunk 4]
```diff
@@ -506,7 +506,9 @@ const styles = StyleSheet.create({
     top: '-13%',
     left: 0,
     right: 0,
+    flex: 1,
```
Comment (tstirrat) [inline app/views/LocationTracking.js:552]: This looks like an unrelated change, what does it do? Perhaps you can revert this change if it's not related.

File: app/components/TrustedSourceList.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  selectedAuthorities,
+  authoritiesList,
+  isAuthorityFilterActive,
+  toggleFilterAuthoritesByGPSHistory,
+  addAuthorityToState,
+  filterAuthoritesByGPSHistory,
+  onMenuClick,
+}) => {
+  const isAuthoritySelected = name => {
+    return findIndex(selectedAuthorities, ['key', name]) < 0;
```
Comment (tstirrat) [inline app/components/TrustedSourceList.js:22]:
```suggestion
    return findIndex(selectedAuthorities, {key: name}) < 0;
```

This is a little more clear

File: app/components/TrustedSourceList.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+          </Typography>
+          <Switch
+            onValueChange={val => filterAuthoritesByGPSHistory({ val })}
+            value={isAuthorityFilterActive}
+          />
+        </TouchableOpacity>
+      )}
+      {authoritiesList === undefined
+        ? null
+        : authoritiesList.map(item => {
```
Comment (tstirrat) [inline app/components/TrustedSourceList.js]: There's a lot goin on in this block. Hard to follow. 

This is `.filter()` operation:

```js
const availableAuthorities = authoritiesList.filter(item => !isAuthoritySelected(item));
```

do that above, and then it makes this much clearer:

```js
{availableAuthorities && availableAuthorities.map(item => <MenuOption ... />)}
```

File: app/components/TrustedSourceList.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
+import findIndex from 'lodash/findIndex';
+import React from 'react';
+import { StyleSheet, Switch, TouchableOpacity } from 'react-native';
+import { MenuOption } from 'react-native-popup-menu';
+
+import { Typography } from '../components/Typography';
+import Colors from '../constants/colors';
+import languages from '../locales/languages';
```
Comment (tstirrat) [inline app/components/TrustedSourceList.js]: use `const {t} = useTranslation()` hook instead

File: app/components/TrustedSourceList.js (hunk 5 above)
Comment (tstirrat) [reply app/components/TrustedSourceList.js:22]: I realise due to the format of the authorities list, this will not work :) never mind.

**Comment by Shabanisimo** [comment]
Updated. I also skipped it right away, but now I fixed the bug with the adding of custom links.

File: app/components/TrustedSourceList.js [hunk 8]
```diff
@@ -16,9 +16,14 @@ export const TrustedSourceList = ({
   filterAuthoritesByGPSHistory,
   onMenuClick,
 }) => {
-  const isAuthoritySelected = name => {
+  const { t } = useTranslation();
+  const isAuthoritySelected = item => {
```
Comment (tstirrat) [inline app/components/TrustedSourceList.js:20]: `isAuthoritySelected` seems to be the opposite of the name..

It returns true if the authority is NOT found in the list.

Either change name to `isAuthorityAvailable`, or change the return value to be `return findIndex(...) >= 0`

File: app/components/TrustedSourceList.js [hunk 9]
```diff
@@ -27,41 +32,35 @@ export const TrustedSourceList = ({
...
           />
         </TouchableOpacity>
       )}
-      {authoritiesList === undefined
-        ? null
-        : authoritiesList.map(item => {
-            let name = Object.keys(item)[0];
-            let key = authoritiesList.indexOf(item);
+      {availableAuthorities &&
+        availableAuthorities.map(item => {
```
Comment (tstirrat) [inline app/components/TrustedSourceList.js:44]: create a unit test for this to show that it renders the correct lists depending on availableAuthorities and selectedAuthorities.
